frontend/ui/network/manager.lua

```diff
@@ -60,6 +61,39 @@ function NetworkMgr:promptWifiOff(complete_callback)
     })
 end
 
+function NetworkMgr:turnOnWifiAndWaitForConnection(callback)
+    NetworkMgr:turnOnWifi()
+    local timeout = 30
+    local retry_count = 0
+    local info = InfoMessage:new{ text = T(_("Enabling Wi-Fi. Waiting for Internet connection…\nTimeout %1 seconds."), timeout)}
+    UIManager:show(info)
+    UIManager:forceRePaint()
+    while not NetworkMgr:isOnline() and retry_count < timeout do
+        FFIUtil.sleep(1)
+        retry_count = retry_count + 1
+    end
+    UIManager:close(info)
+    if retry_count == timeout then
+        UIManager:show(InfoMessage:new{ text = _("Error connecting to the network") })
+        return
+    end
+    if callback then callback() end
+end
+
+function NetworkMgr:wifiEnableAction(callback)
```

Comment:
The naming convention here is a bit of a mess. It should be `enableWifiAction` if it is to keep this name, but I'm not sure if this function should be a separate thing. My personal inclination is to replace `promptWifiOn` with this.

Pinging @poire-z for a second opinion.

Reply:
I think for cleaner code would be better if we have separate function, 
& apply "search all" for "promptWifiOn" and replace them with `enableWifiAction` (or whatever you want call it). 
I could do that (simple FindAll in IDE;).

(As `promptWifiOn` will be misleading after adding option to "silent enable wifi")

Reply:
Separate function is not related in the sense I intended. You split up code with separate local functions, inaccessible from the outside.

Anyway, so currently we have:

* `NetworkMgr:turnOnWifi()` (device specific; not really to be used elsewhere…)
* `NetworkMgr:promptWifiOn(complete_callback)` (method to be called by the program; the prompt can optionally be disabled by the user)
* `NetworkMgr:enableWifiAction(callback)` (enable/turnOn discrepancy, should that action thing even be in there…)

I think the way out of the mess is something like this:

1. change `turnOnWifi` to `turnOnWifiHW` (by analogy with `frontlightIntensityHW`)
2. use the now free `turnOnWifi` for what is currently called `enableWifiAction`
3. we keep `promptWifiOn` as a method but mass rename references to `turnOnWifi`

Reply:
I have given my opinion in https://github.com/koreader/koreader/pull/3395#issuecomment-357796949

Reply:
Also note, and I apologize for that, but I haven't really given much look at the NewsDownloaded PRs - so if some stuff I suggested have already been done around them, ignore them.

Reply:
Is there anything to be implemented in this pr from the discussion in #3395? 
Or if will be in other pr? 
Cause I don't quite follow

Reply:
That's pretty much all for the future. If you change the name of `wifiEnableAction` to `beforeWifiAction` that should about do the trick. :+1: